### PR TITLE
Added possibility to use env vars on disk path and url

### PR DESCRIPTION
**Pull request by david-caro**
Also refactored a bit the image and disk handling code.

Change-Id: I286c275a17ae9116fc606d2e2bdc92bccf62cf64
Signed-off-by: David Caro <dcaroest@redhat.com>

### CODE REVIEW

**Comment by david-caro** [comment]
ci :shipit:

File: lago/prefix.py [hunk 1]
```diff
@@ -451,102 +460,188 @@ def _create_disk(
...
-                    qemu_img_cmd = [
-                        'qemu-img', 'create', '-f', 'qcow2', '-b', base,
-                        disk_path
-                    ]
+        if url:
+            disk_spec['path'] = self._retrieve_disk_url(url, disk_path)
+        else:
+            disk_spec['path'] = disk_path
+
+        # If we're using raw file, return it's path
```
Comment (mykaul) [inline lago/prefix.py]: it's -> its

File: lago/prefix.py [hunk 2]
```diff
@@ -451,102 +460,188 @@ def _create_disk(
...
-                        'qemu-img', 'create', '-f', 'qcow2', '-b', path,
-                        disk_path
-                    ]
-                else:
-                    raise RuntimeError(
-                        'Unsupporte template spec %s' % str(spec)
-                    )
-                task_message = 'Create disk %s(%s)' % (name, spec['name'])
+        qemu_cmd = [
+            'qemu-img', 'create', '-f', disk_spec['format'], disk_path,
```
Comment (mykaul) [inline lago/prefix.py]: If the disk format is qcow2, I suggest adding '-o preallocation=metadata' which should improve read performance very nicely. It cannot be combined with a baking file though. 
1. I wonder how many reads we do from a non-backed qcow2 file.
2. We should ensure our lago-images created images Are created with 'preallocation=metadata'.

File: lago/prefix.py [hunk 3]
```diff
@@ -451,102 +460,188 @@ def _create_disk(
...
+        with LogTask(
+            'Create empty disk %s(%s)' % (host_name, disk_spec['name'])
+        ):
+            ret, _, _ = utils.run_command(qemu_cmd)
+            if ret != 0:
+                raise RuntimeError(
+                    'Failed to create image, qemu-img returned %d' % ret,
+                )
+            # To avoid losing access to the file
+            os.chmod(disk_path, 0666)
```
Comment (mykaul) [inline lago/prefix.py]: why is that red?  Devilish?

Comment (david-caro) [inline lago/prefix.py]: heheh, I thought exactly the same :)

File: lago/prefix.py (hunk 2 above)
Comment (david-caro) [inline lago/prefix.py]: I've already started the merge gate for this change (minutes before your first comment), I'll open a new one with the new changes:

    https://github.com/lago-project/lago/issues/254

**Comment by david-caro** [comment]
ci :shipit:

**Pull request merged**